src/main/java/fun/epak/pak/service/UserService.java

```diff
@@ -106,21 +103,40 @@ public void saveChangedUser(ChangeUserDataRequest userData, MultipartFile multip
     public OtherUserProfileData loadOtherUserProfileData(String email, long id) {
         User user = userRepository.findById(id).orElseThrow();
         User viewer = userRepository.findByEmail(email).orElseThrow();
+        boolean contains = viewer.getSubscriptions().contains(user);
         String userImagePath = imageBaseAddress + user.getId() + "/" + user.getImageName();
-        return OtherUserProfileData.of(user, userImagePath, viewer);
+        return OtherUserProfileData.of(user, userImagePath, contains);
     }
 
-    public void changeUserSubscriptionList(String email, long id) {
-        User user = userRepository.findByEmail(email).orElseThrow();
-        User subscriptedUser = userRepository.findById(id).orElseThrow();
-        if (user.equals(subscriptedUser)) {
+    public void subscribeUser(String subscribingEmail, long subscribedId) {
+        User user = userRepository.findByEmail(subscribingEmail).orElseThrow();
+        User subscribedUser = userRepository.findById(subscribedId).orElseThrow();
+        if (user.equals(subscribedUser)) {
             throw new SubscribeYourselfException("You cant subscribe to yourself");
         }
-        if (user.getSubscriptions().contains(subscriptedUser)) {
-            user.getSubscriptions().remove(subscriptedUser);
-        } else {
-            user.getSubscriptions().add(subscriptedUser);
+        user.getSubscriptions().add(subscribedUser);
+        userRepository.save(user);
+    }
+
+    public void unsubscribeUser(String unsubscribingEmail, long unsubscribedId) {
+        User user = userRepository.findByEmail(unsubscribingEmail).orElseThrow();
+        User subscribedUser = userRepository.findById(unsubscribedId).orElseThrow();
+        if (user.equals(subscribedUser)) {
```

Comment:
[Gatunek optymalizacja] to sprawdzenie może być przed kolejnym zmęczeniem bazy pytaniem (wyżej też):
```
if(user.getId() == unsubscribedId) {
...
}
```